lib/rubocop/cop/minitest/assert_nil.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # Check if your test uses `assert_nil` instead of `assert_equal(nil, something)`.
+      #
+      # @example
+      #   # bad
+      #   assert_equal(nil, actual)
+      #   assert_equal(nil, actual, 'the message')
+      #
+      #   # good
+      #   assert_nil(actual)
+      #   assert_nil(actual, 'the message')
+      #
+      class AssertNil < Cop
+        MSG = 'Prefer using `assert_nil(%<arguments>s)` over ' \
+              '`assert_equal(nil, %<arguments>s)`.'
+
+        def_node_matcher :assert_equal_with_nil, <<~PATTERN
+          (send nil? :assert_equal nil (... $_) (str $_)? )
```

Comment:
I noticed two points of concern in the node pattern.

First, false negatives occur when a method is called as follows:

```ruby
assert_equal(nil, obj.do_something, 'message')
```

A test case will be added for the cop and the following changes will be required.

```diff
--- a/lib/rubocop/cop/minitest/assert_nil.rb
+++ b/lib/rubocop/cop/minitest/assert_nil.rb
@@ -19,7 +19,7 @@ module RuboCop
               '`assert_equal(nil, %<arguments>s)`.'

         def_node_matcher :assert_equal_with_nil, <<~PATTERN
-          (send nil? :assert_equal nil (... $_) (str $_)? )
+          (send nil? :assert_equal nil $_ (str $_)? )
         PATTERN

         def on_send(node)
@@ -30,7 +30,7 @@ module RuboCop
                                   assertion_message.first
                                 end

-            arguments = [argument, assertion_message].compact.join(', ')
+            arguments = [argument.source, assertion_message].compact.join(', ')

             add_offense(node, message: format(MSG, arguments: arguments))
           end
```

Second, the third argument of `assert_equal` can accept anything other than a string.
For example, there are false negatives in the following cases.

```ruby
assert_equal(nil, obj, message)
assert_equal(nil, obj, MESSAGE)
```

The node pattern will also need to be changed.

Can you change this?